pkg/service/claimlookup/claimlookup.go

```diff
@@ -1,15 +1,66 @@
 package claimlookup
 
 import (
-	"github.com/ipni/go-libipni/find/model"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/ipfs/go-cid"
 	"github.com/storacha-network/go-ucanto/core/delegation"
+	"github.com/storacha-network/indexing-service/pkg/types"
 )
 
-// ClaimLookup is used to get full claims from a claim cid
-// I'll be honest, I'm not exactly sure whether these claims should be stored or simply synthesized
-// from the information in IPNI combined with having private keys stored in this service
-// I THINK it's possible you can synthesize Index & Equals claims from the information in IPNI + a private key
-// Location commitments are more complicated cause they really ought to be signed by the storer of the commitment?
-type ClaimLookup interface {
-	LookupClaim(model.ProviderResult) (delegation.Delegation, error)
+// ClaimLookup is a read through cache for fetching content claims
+type ClaimLookup struct {
+	httpClient *http.Client
+	claimStore types.ContentClaimsStore
+}
+
+// NewClaimLookup creates a new ClaimLookup with the provided claimstore and HTTP client
+func NewClaimLookup(claimStore types.ContentClaimsStore, httpClient *http.Client) *ClaimLookup {
+	return &ClaimLookup{
+		httpClient: httpClient,
+		claimStore: claimStore,
+	}
+}
+
+// LookupClaim attempts to fetch a claim from either the local cache or via the provided URL (caching the result if its fetched)
```

Comment:
suggestion: personally I'd implement the logic for caching in both `ClaimLookup` and `BlobIndexLookup` as wrappers around the fetchers that expose the same API, so you can test in isolation, and not (for example) have to mock out a `claimStore` to test the fetching side...or easily mock the fetcher when your testing the caching side.

Something like:

```go
lookup := claimlookup.NewClaimLookup(httpClient)
cachedLookup := claimlookup.WithCache(lookup, claimStore)

cachedLookup.LookupClaim(/* ... */)
```